### PR TITLE
emulator name was undefined because we missed an await

### PR DESCRIPTION
**Pull request by Plamen5kov**
See more info: https://github.com/NativeScript/nativescript-cli/issues/2576

_Problem_
When `tns run/debug android` is ran, with no running emulator, the emulator was started, the app was installed, but not launched. The problem is after the emulator was started we couldn't get the name of the emulator.

Related issue: https://github.com/NativeScript/nativescript-cli/issues/2486

### CODE REVIEW

**Comment by justcodebuilduser** [comment]
:heart:

File: mobile/android/android-emulator-services.ts [hunk 1]
```diff
@@ -184,7 +184,7 @@ class AndroidEmulatorServices implements Mobile.IAndroidEmulatorServices {
 
 		// if we get here, there's at least one running emulator
 		let getNameFunction = this.$options.geny ? this.getNameFromGenymotionEmulatorId : this.getNameFromSDKEmulatorId;
-		let emulatorId = await _(runningEmulators).find(emulator => getNameFunction.apply(this, [emulator]) === image);
+		let emulatorId = await _(runningEmulators).find(async emulator => await getNameFunction.apply(this, [emulator]) === image);
```
Comment (rosen-vladimirov) [inline mobile/android/android-emulator-services.ts]: this code will not wait for the execution of the lambdas inside `find`
It should be something like:
```TypeScript
for (let emulator of runningEmulators) {
    const currentEmulatorName = await getNameFunction.apply(this, [emulator]);
    if (currentEmulatorName === image) {
        return emulator;
    }
}
```

Comment (Plamen5kov) [inline mobile/android/android-emulator-services.ts]: As I looked at the generated javascript, I came to the same conclusion. Will fix.

**Comment by justcodebuilduser** [comment]
:heart:

File: mobile/android/android-emulator-services.ts [hunk 2]
```diff
@@ -184,9 +184,13 @@ class AndroidEmulatorServices implements Mobile.IAndroidEmulatorServices {
...
-		let emulatorId = await _(runningEmulators).find(emulator => getNameFunction.apply(this, [emulator]) === image);
-
-		return emulatorId;
+		for (let emulatorId of runningEmulators) {
+			const currentEmulatorName = await getNameFunction.apply(this, [emulatorId]);
+			if (currentEmulatorName === image) {
+				return emulatorId;
+			}
+		}
+		this.$errors.failWithoutHelp("Couldn't find emulator id");
```
Comment (rosen-vladimirov) [inline mobile/android/android-emulator-services.ts]: This changes the behavior of the method. Previously it's been returning undefined. And the call to the method relies on this undefined value:
https://github.com/telerik/mobile-cli-lib/blob/master/mobile/android/android-emulator-services.ts#L269

Is there a particular reason to add this failure here?

Comment (Plamen5kov) [inline mobile/android/android-emulator-services.ts]: Forgot it there, while testing.

**Comment by justcodebuilduser** [comment]
:heart:

**Comment by Plamen5kov** [comment]
related to: https://github.com/NativeScript/nativescript-cli/issues/2210

>It happen always when emulator is not started before tns run android
this is fixed.

**Pull request merged**